src/printer.js

```diff
@@ -626,7 +628,7 @@ function genericPrintNoParens(path, options, print) {
           concat([
             leftBrace,
             indent(
-              options.tabWidth,
+              options.tabWidth * (parentIsUnionTypeAnnotation ? 2 : 1),
```

Comment:
To indent the end brace, just to a similar thing below around `rightBrace`:

`parentIsUnionTypeAnnotation ? indent(3, rightBrace), rightBrace`

I could be wrong, but something like that should work.